WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostCardActionsHandler.kt

```diff
@@ -121,16 +135,43 @@ class ReaderPostCardActionsHandler @Inject constructor(
         _navigationEvents.postValue(Event(OpenPost(post)))
     }
 
-    private fun handleBlockSiteClicked(postId: Long, blogId: Long) {
-        AppLog.d(AppLog.T.READER, "Block site not implemented")
+    private fun handleBlockSiteClicked(blogId: Long) {
+        defaultScope.launch {
+            blockBlogUseCase.blockBlog(blogId).collect {
+                when (it) {
+                    is SiteBlockedInLocalDb -> {
+                        _refreshPosts.postValue(Event(Unit))
+                        _snackbarEvents.postValue(
+                                Event(
+                                        SnackbarMessageHolder(string.reader_toast_blog_blocked, string.undo, {
+                                            defaultScope.launch {
+                                                ReaderBlogActions.undoBlockBlogFromReader(it.blockedBlogData)
```

Comment:
We could move the "undo" action into it's own usecase, but it felt like an overkill since it's a fire&forget action.

Reply:
> it felt like an overkill 
I just hope it doesn't become a wrong precedent in the future.

Reply:
If you prefer, I can wrap it into a use case. It won't hurt I guess. Wdyt?

Reply:
Yep lets wrap it in usecase

Reply:
Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/12792/commits/be6113a938999241157e5ccd54667fbbc65a7c42